core/src/main/scala/org/apache/spark/rpc/akka/AkkaRpcEnv.scala

```diff
@@ -237,6 +237,12 @@ private[spark] class AkkaRpcEnv private[akka] (
   }
 
   override def toString: String = s"${getClass.getSimpleName}($actorSystem)"
+
+  def deserialize[T](deserializationAction: () => T): T = {
+    JavaSerializer.currentSystem.withValue(actorSystem.asInstanceOf[ExtendedActorSystem]) {
```

Comment:
spark JavaSerializer is used to deserialize objects. However, it does not have an actor system in the current context. I need to use Akka `JavaSerializer.currentSystem` to put the current actor system into a thread-local variable.